docs/admin_api/user_admin_api.rst

```diff
@@ -823,3 +823,114 @@ The following parameters should be set in the URL:
 
 - ``user_id`` - The fully qualified MXID: for example, ``@user:server.com``. The user must
   be local.
+
+Override ratelimiting for users
+===============================
+
+This API allows to override or disable rate limiting for a specific user.
+There are specific APIs to set, get and delete a ratelimit.
+
+Get status of ratelimit
+-----------------------
+
+The API is::
+
+  GET /_synapse/admin/v1/users/<user_id>/override_ratelimit
+
+To use it, you will need to authenticate by providing an ``access_token`` for a
+server admin: see `README.rst <README.rst>`_.
+
+A response body like the following is returned:
+
+.. code:: json
+
+    {
+      "messages_per_second": 0,
+      "burst_count": 0
+    }
+
+**Parameters**
+
+The following parameters should be set in the URL:
+
+- ``user_id`` - The fully qualified MXID: for example, ``@user:server.com``. The user must
+  be local.
+
+**Response**
+
+The following fields are returned in the JSON response body:
+
+- ``messages_per_second`` - integer - The number of actions that can
+  be performed in a second.
+- ``burst_count`` - integer - How many actions that can be performed
+  before being limited.
```

Comment:
Can we document that `0` or `null` mean ratelimiting is disabled? Or we should replace `0` with `null` (or vice versa) before returning the values, for consistency sake?

Reply:
If i read the code correctly, then `null` is better for deactivating, as it breaks the processing sooner!?

https://github.com/matrix-org/synapse/blob/5b268997bdde04c905eed0a7c04e9e2352e35fba/synapse/handlers/_base.py#L102-L108

Reply:
`if not override.messages_per_second:` and `if not override.messages_per_second or override.messages_per_second == 0:` are equivalent here (since both `None` and `0` are falsey in python), which is why we've ended up in a situation where `0` and `None` mean the same thing anyway.

Reply:
I think the only thing we might want to do here is to change `null` to `0` (or vice versa) when we return the values from the API, for consistency. I don't really mind though.

(The fact that we treat `0` and `null` the same is a bit awful, but untangling that may be a bit of a mess)

Reply:
You can probably just cast to an int to always return `0`?

Reply:
Yes, I can do it.
Should I do the conversion from `None` to `0` only in admin API or direct in store `get_ratelimit_for_user`?

Reply:
I'd probably do it in the API for now, just to avoid unnecessarily changing things used by lots of places.